src/sentry/api/endpoints/organization_artifactbundle_assemble.py

```diff
@@ -77,13 +80,25 @@ def post(self, request: Request, organization) -> Response:
 
         from sentry.tasks.assemble import assemble_artifacts
 
+        version = data.get("version")
+        dist = data.get("dist")
+
+        if version is None and dist is not None:
+            return Response(
+                {"error": "You need to specify a version together with a dist"}, status=400
```

Comment:
I agree this is super confusing, I will take care of refactoring it across all the code. I was using `version` since it was already present in async job params. I will keep it as version there to avoid issues of old code calling with keyword version on the new code.